pkg/config/ext_authz_test.go

```diff
@@ -0,0 +1,316 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	extAuthService "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/ext_authz/v3"
+	httpOptions "github.com/envoyproxy/go-control-plane/envoy/extensions/upstreams/http/v3"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+func Test_isValidProtocol(t *testing.T) {
+	tests := []struct {
+		protocol extAuthzProtocol
+		want     bool
+	}{
+		{protocol: "grpc", want: true},
+		{protocol: "http", want: true},
+		{protocol: "https", want: true},
+		{protocol: "", want: false},
+		{protocol: "unknown", want: false},
+	}
+	for _, tt := range tests {
+		t.Run(string(tt.protocol), func(t *testing.T) {
+			if got := isValidExtAuthzProtocol(tt.protocol); got != tt.want {
+				t.Errorf("isValidExtAuthzProtocol() = %v, extAuthzWanted %v", got, tt.want)
```

Comment:
```suggestion
				t.Errorf("isValidExtAuthzProtocol() = %v, want = %v", got, tt.want)
```

serving repo uses this style, so let's align with them.